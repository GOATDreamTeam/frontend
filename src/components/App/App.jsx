import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route
} from 'react-router-dom';
require ('./App.css');
import Header from '../header/Header';
import SplashPage from '../splash-page/SplashPage';
// import Footer from '../footer/Footer';
import List from '../list/List';
import DetailPage from '../detail/DetailPage';
import AboutUs from '../aboutus/AboutUs';
import { useAccessibility } from '../../hooks/appContext';
import { useStyles } from '../../hooks/globalStyles';


export default function App() {

  const { theme } = useAccessibility();
  const { basicDisplay } = useStyles(theme);
  
  return (
    <div className={basicDisplay}>

      <Router>
        <Header/>
        <Switch>
          <Route exact path="/" component={SplashPage} />
          <Route path="/plants-list/:plantName" component={List} />
          <Route path="/plant-detail/:id" component={DetailPage} />
          <Route path="/about-us" component={AboutUs} />
        </Switch>
      </Router>

    </div>
    
  );
}
  
