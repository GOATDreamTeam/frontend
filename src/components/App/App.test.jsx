import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import AccessibilityProvider from '../../providers/AppProvider';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<AccessibilityProvider><App /></AccessibilityProvider>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
