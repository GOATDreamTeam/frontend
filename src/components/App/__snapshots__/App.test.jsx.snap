// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<BrowserRouter>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/plants-list/:plantName"
    />
    <Route
      component={[Function]}
      path="/plant-detail/:id"
    />
    <Route
      component={[Function]}
      path="/about-us"
    />
  </Switch>
  <Footer />
</BrowserRouter>
`;
